core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,52 @@
+{% macro get_show_grant_sql(relation) %}
+{{ return(adapter.dispatch("get_show_grant_sql", "dbt")(relation)) }}
+{% endmacro %}
+
+{% macro default__get_show_grant_sql(relation) %}
+show grants on {{ relation.type }} {{ relation }}
+{% endmacro %}
+
+{% macro get_grant_sql(relation, grant_config) %}
+{{ return(adapter.dispatch('get_grant_sql', 'dbt')(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_grant_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+     {{ log('privilege: ' ~ privilege) }}
+        {% set grantees = grant_config[privilege] %}
+        {% for grantee in grantees %}
+            grant {{ privilege }} on {{ relation.type }} {{ relation }} to {{ grantee }};
+        {% endfor %}
+    {% endfor %}
+{% endmacro %}
+
+{% macro get_revoke_sql(relation, grant_config) %}
+{{ return(adapter.dispatch("get_revoke_sql", "dbt")(relation, grant_config)) }}
+{% endmacro %}
+
+{% macro default__get_revoke_sql(relation, grant_config) %}
+    {% for privilege in grant_config.keys() %}
+        {% set grantees = grant_config[privilege] %}
+        {% for grantee in grantees if grantee !=  target.user %}
```

Comment:
Ah! I don't know if this is the spot where we want to filter out `grantee != target.user`. I would rather have that check during the `show grants` / post-processing, so that we don't even consider the current user's grants to begin with.

As in `get_grant_sql` above, we can also avoid the extra loop step and just join `grantees` together as a comma-separated list.